dcc-portal-ui/app/scripts/repository/js/repository.js

```diff
@@ -290,6 +290,40 @@ import './file-finder';
       jQuery(tween.elem).closest('td').removeClass('tweening');
     };
 
+    $scope.isGeneratingIcgcGetId = false;
+    $scope.icgcGetId = null;
+    let abortGenerateIcgcGetId = false;
+
+    const generateIcgcGetId = (repoOrder) => {
+      $scope.isGeneratingIcgcGetId = true;
+      $timeout(() => {
+        if (abortGenerateIcgcGetId) {
+          abortGenerateIcgcGetId = false;
+        } else {
+          $scope.isGeneratingIcgcGetId = false;
+          $scope.icgcGetId = '594ae01c-52e3-4488-9602-ec2a3f416dd9';
```

Comment:
@cheapsteak the manifestId with repo order preserved can be generated using
`curl -X POST --header 'Content-Type: application/x-www-form-urlencoded'   --header 'Accept: text/plain' -d "repos=$urlEncodedRepos\&filters=$urlEncodedFilters"    'https://dcc.icgc.org/api/v1/manifests' `

where the variable `$urlEncodedRepos` is an url encoded string containing a list of repo names (this is where you can change the order or priority of repositories) joined by a comma, and the variable `$urlEncodedFilters` is an url encoded string containing the filter for the file repository search. the response of this request is a manifestId